esmf_regrid/_esmf_sdo.py

```diff
@@ -275,3 +290,164 @@ def _make_esmf_sdo(self):
+
+    This class holds information about lat-lon type grids. That is, grids
+    defined by lists of latitude and longitude values for points/bounds
+    (with respect to some coordinate reference system i.e. rotated pole).
+    It contains methods for translating this information into :mod:`ESMF` objects.
+    In particular, there are methods for representing as a
+    :class:`ESMF.api.grid.Grid` and
+    as a :class:`ESMF.api.field.Field` containing that
+    :class:`~ESMF.api.grid.Grid`. This ESMF :class:`~ESMF.api.field.Field`
+    is designed to be a higher resolution version of the given grid and should
+    contain enough information for area weighted regridding though this may be
+    inappropriate for other :mod:`ESMF` regridding schemes.
+
+    """
+
+    def __init__(
+        self,
+        lonbounds,
+        latbounds,
+        resolution=400,
+        crs=None,
+    ):
+        """
+        Create a :class:`GridInfo` object describing the grid.
+
+        Parameters
+        ----------
+        lons : :obj:`~numpy.typing.ArrayLike`
+            A 1D or 2D array or list describing the longitudes of the
+            grid points.
+        lats : :obj:`~numpy.typing.ArrayLike`
+            A 1D or 2D array or list describing the latitudes of the
+            grid points.
+        lonbounds : :obj:`~numpy.typing.ArrayLike`
+            A 1D or 2D array or list describing the longitude bounds of
+            the grid. Should have length one greater than ``lons``.
+        latbounds : :obj:`~numpy.typing.ArrayLike`
+            A 1D or 2D array or list describing the latitude bounds of
+            the grid. Should have length one greater than ``lats``.
+        resolution : int, default=400
+            A number describing how many latitude slices each cell should
+            be divided into when passing a higher resolution grid to ESMF.
+        crs : :class:`cartopy.crs.CRS`, optional
+            Describes how to interpret the
+            above arguments. If ``None``, defaults to :class:`~cartopy.crs.Geodetic`.
+
+        """
+        self.resolution = resolution
+        self.n_lons_orig = len(lonbounds) - 1
+        self.n_lats_orig = len(latbounds) - 1
+
+        # Create dummy lat/lon values
+        lons = np.zeros(self.n_lons_orig)
+        lats = np.zeros(self.n_lats_orig)
+        super().__init__(lons, lats, lonbounds, latbounds, crs=crs)
+
+        if self.n_lats_orig == 1 and np.allclose(latbounds, [-90, 90]):
+            self._refined_latbounds = np.array([-90, 0, 90])
+            self._refined_lonbounds = lonbounds
+            self.lon_expansion = 1
+            self.lat_expansion = 2
+        else:
+            self._refined_latbounds = latbounds
+            self._refined_lonbounds = np.append(
+                np.linspace(
+                    lonbounds[:-1],
+                    lonbounds[1:],
+                    self.resolution,
+                    endpoint=False,
+                    axis=1,
```

Comment:
I think ESMF is able to handle this stuff fine, it's just that it would require a bit too much fuss on our end to figure out how to handle such cases, so for now it's probably safer to explicitly disallow them.